src/App.tsx

```diff
@@ -47,9 +51,152 @@ function App() {
 
     }, [selectedFile]);
 
+
+    function isLastDay(date: string) {
+        let d = new Date(date)
+        let test = new Date(d.getTime())
+        let month = test.getMonth();
+        test.setDate(test.getDate() + 1);
+        return test.getMonth() !== month;
+    }
+
+
+    const sortedData = useMemo(() => {
+
+        if (sortBy === 'daily') {
+            return data
+        } else {
+            let total_days = 0
+            let week_day = 0
+            let pr_sum = 0
+            let temp_max = 0
+            let temp_min = 0
+            let wind_sum = 0
+            let arr: IDataShape[] = []
+            let weather = {
+                fog: 0,
+                rain: 0,
+                snow: 0,
+                drizzle: 0,
+                sun: 0,
+            }
+
+            if (sortBy === 'weekly') {
+                for (let i = 0; i < data.length; i++) {
+                    total_days += 1
+                    week_day = Number(dayjs(data[i].date).format('d'))
+                    pr_sum += data[i].precipitation
+                    temp_max += data[i].temp_max
+                    temp_min += data[i].temp_min
+                    wind_sum += data[i].wind
+                    // @ts-ignore
+                    weather[data[i].weather] += 1
+
+                    if (week_day === 0) {
+                        // @ts-ignore
+                        // @ts-ignore
+                        // @ts-ignore
+                        // @ts-ignore
+                        // @ts-ignore
+                        arr.push({
+                            date: data[i].date,
+                            precipitation: Number((pr_sum / total_days).toFixed(2)),
+                            temp_max: Number((temp_max / total_days).toFixed(2)),
+                            temp_min: Number((temp_min / total_days).toFixed(2)),
+                            wind: Number((wind_sum / total_days).toFixed(2)),
+                            // @ts-ignore
+                            weather:  Object.keys(weather).reduce((a, b) => (weather[a] > weather[b] ? a : b)),
+                        });
+                        total_days = 0;
+                        pr_sum = 0.0;
+                        temp_max = 0.0;
+                        temp_min = 0.0;
+                        wind_sum = 0.0;
+                        weather.rain = 0;
+                        weather.snow = 0;
+                        weather.drizzle = 0;
+                        weather.sun = 0;
+                        weather.fog = 0;
+                    }
+                }
+            } else {
+                for (let i = 0; i < data.length; i++) {
+                    total_days += 1
```

Comment:
I have seen there is a last empty line which has invalida data, so I vaildated this here and checked if it is an empty string.

```
            if (0 === data[i].date.length) {
                continue;
            }
```